### PR TITLE
Mapper support set device status reportcycle

### LINKED ISSUE #5892: Mapper supports setting devicestatus reporting frequency
**What would you like to be added/modified**:

Users can set whether to report device status to the cloud in device CRD, and set the frequency of reporting.

**Why is this needed**:

In 1.18, we implemented the feature of device status collection and reporting. But currently the device status is reported to the cloud by default, and the frequency of device status reporting cannot be controlled through device CRD. This will result in frequent output of logs reporting device status when debugging and using mapper, and the default reporting cycle may cause too many requests error.  So we need to add the `reportToCloud` flag to control whether the device status is reported to the cloud, and users can also control the status reporting cycle through device CRD.

自动上报status

status太多

### PR DESCRIPTION
**Pull request by wbc6080**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:

In #5892, we find the device status is reported to the cloud by default currently, and the frequency of device status reporting cannot be controlled through device CRD. In this PR, we add the `reportToCloud` and `reportCycle` field in `device.status` of device-instance crd, allowing users to set whether to push the device status to the cloud and the push frequency in the device-instance yaml.

**Which issue(s) this PR fixes**:

Fixes #5892

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by wbc6080** [comment]
In the original `device-instance CRD`, user can set whether to report to the cloud and the reporting frequency in each device property. 
image

Since the reporting frequency of each device property may be different, device status cannot reuse the value set in the device property.In this PR, we set these two fields separately for the device status in the `device-instance CRD` to control whether the device status is reported to the cloud and the frequency of reporting. An example configuration file of device-instance after adding them is as follows:
image

**Comment by wbc6080** [comment]
Ready for review. 
/assign @luomengY @JiaweiGithub @WillardHu

**Comment by luomengY** [comment]
`ReportToCloud` seems to already have this feature.

**Comment by luomengY** [comment]
image

**Comment by wbc6080** [comment]
> `ReportToCloud` seems to already have this feature.

This is true, but this field is used to control device property and cannot be used directly to control the reporting of device status.

File: staging/src/github.com/kubeedge/api/apis/dmi/v1beta1/api.proto [hunk 1]
```diff
@@ -367,6 +367,11 @@ message ReportDeviceStatesRequest {
 message DeviceStatus {
     // the device twins of the device.
     repeated Twin twins = 1;
+    // whether be reported to the cloud
+    bool ReportToCloud = 2;
```
Comment (fisherxu) [inline staging/src/github.com/kubeedge/api/apis/dmi/v1beta1/api.proto:371]: Start with the lowercase as below?

**Review by fisherxu — COMMENTED** [review]
/approve
cc @WillardHu @JiaweiGithub

File: staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/devicestatus.go [hunk 2]
```diff
@@ -49,15 +51,22 @@ func (deviceStates *DeviceStates) PushStatesToEdgeCore() {
...
-	ticker := time.NewTicker(2 * time.Second)
+	// No need to report device status to the cloud
+	if !deviceStates.ReportToCloud {
+		return
+	}
+	// Set device status report cycle
+	if deviceStates.ReportCycle == 0 {
+		deviceStates.ReportCycle = common.DefaultReportCycle
+	}
+	ticker := time.NewTicker(deviceStates.ReportCycle)
```
Comment (WillardHu) [inline staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/devicestatus.go:69]: Define a minimum unit of time. `deviceStates.ReportCycle*time.Second` or `deviceStates.ReportCycle*time.Millisecond`

**Comment by WillardHu** [comment]
After this PR is merged, will the other one https://github.com/kubeedge/kubeedge/pull/5885 not be needed?

File: staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/devicestatus.go (hunk 2 above)
Comment (wbc6080) [reply staging/src/github.com/kubeedge/mapper-framework/_template/mapper/device/devicestatus.go:69]: We have defined the time unit during initialization
image

**Pull request merged**